File: NOVEMBER-2025/01-11-2025/Course Schedule II.js
class Solution {
    findOrder(n, prerequisites) {
        const adj = Array.from({ length: n }, () => []);
        const indegree = Array(n).fill(0);
        for (let [course, prereq] of prerequisites) {
            adj[prereq].push(course);
            indegree[course]++;
        }
        const queue = [];
        for (let i = 0; i < n; i++) {
            if (indegree[i] === 0) queue.push(i);
        }
        const order = [];
        while (queue.length > 0) {
            const curr = queue.shift();
            order.push(curr);
            for (let next of adj[curr]) {
                indegree[next]--;
                if (indegree[next] === 0) queue.push(next);
            }
        }
        return order.length === n ? order : [];
    }
}

File: NOVEMBER-2025/01-11-2025/Course Schedule II.cs
class Solution {
    public List<int> findOrder(int n, int[,] prerequisites) {
        List<int>[] adj = new List<int>[n];
        for (int i = 0; i < n; i++)
            adj[i] = new List<int>();
        int[] indegree = new int[n];
        int m = prerequisites.GetLength(0);
        for (int i = 0; i < m; i++) {
            int course = prerequisites[i, 0];
            int prereq = prerequisites[i, 1];
            adj[prereq].Add(course);
            indegree[course]++;
        }
        Queue<int> q = new Queue<int>();
        for (int i = 0; i < n; i++)
            if (indegree[i] == 0)
                q.Enqueue(i);
        List<int> order = new List<int>();
        while (q.Count > 0) {
            int curr = q.Dequeue();
            order.Add(curr);
            foreach (int next in adj[curr]) {
                indegree[next]--;
                if (indegree[next] == 0)
                    q.Enqueue(next);
            }
        }
        return (order.Count == n) ? order : new List<int>();
    }
}